src/pages/transactions/create/Create.tsx

```diff
@@ -0,0 +1,170 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { Card, Element } from '@invoiceninja/cards';
+import { InputField, SelectField } from '@invoiceninja/forms';
+import {
+  defaultTransactionProperties,
+  transactionTypes,
+} from 'common/constants/transactions';
+import { useCurrencies } from 'common/hooks/useCurrencies';
+import { useTitle } from 'common/hooks/useTitle';
+import { endpoint } from 'common/helpers';
+import { TransactionInput } from 'common/interfaces/transactions';
+import { Container } from 'components/Container';
+import { DebouncedCombobox } from 'components/forms/DebouncedCombobox';
+import { Default } from 'components/layouts/Default';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useTransactionPages } from '../common/hooks/useTransactionPages';
+import { TransactionValidation } from '../common/validation/ValidationInterface';
+import { request } from 'common/helpers/request';
+import toast from 'react-hot-toast';
+import { useNavigate } from 'react-router-dom';
+import { route } from 'common/helpers/route';
+import { useTransactionValidation } from '../common/hooks/useTransactionValidation';
+
+const Create = () => {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const currencies = useCurrencies();
+  const pages = useTransactionPages();
+  const { documentTitle } = useTitle('new_transaction');
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+  const [errors, setErrors] = useState<TransactionValidation>();
+  const [transaction, setTransaction] = useState<TransactionInput>(
+    defaultTransactionProperties
+  );
+  const { checkValidation, setValidation } = useTransactionValidation();
+
+  const handleChange = (
+    property: keyof TransactionInput,
+    value: unknown
+  ): void => {
+    setTransaction(
+      (transaction) => transaction && { ...transaction, [property]: value }
+    );
+  };
+
+  const onSave = async (event: React.FormEvent<HTMLFormElement>) => {
+    event?.preventDefault();
+    setErrors(undefined);
+    const isFormValid = checkValidation(transaction);
+
+    if (!isFormValid && !errors) {
+      setValidation(transaction, setErrors, errors);
+    }
+
+    if (isFormValid) {
+      setIsSaving(true);
+      const toastId = toast.loading(t('processing'));
+      try {
+        await request('POST', endpoint('/api/v1/bank_transactions'), {
+          ...transaction,
+          amount: +transaction?.amount,
+          base_type: transaction?.base_type === 'deposit' ? 'CREDIT' : 'DEBIT',
+        });
+        toast.success(t('created_transaction'), { id: toastId });
+        setIsSaving(false);
+        navigate(route('/transactions'));
+      } catch (error: any) {
```

Comment:
Please typehint the arguments. No `any`.

Reply:
> Please typehint the arguments. No `any`.

done